app/uk/gov/hmrc/bindingtariffrulingfrontend/connector/model/CaseStatus.scala

```diff
@@ -16,11 +16,47 @@
 
 package uk.gov.hmrc.bindingtariffrulingfrontend.connector.model
 
-import play.api.libs.json.Format
-import uk.gov.hmrc.bindingtariffrulingfrontend.utils.EnumJson
+import play.api.libs.json.*
 
-object CaseStatus extends Enumeration {
-  type CaseStatus = Value
-  val DRAFT, NEW, OPEN, SUPPRESSED, REFERRED, REJECTED, CANCELLED, SUSPENDED, COMPLETED, REVOKED, ANNULLED = Value
-  implicit val format: Format[CaseStatus.Value] = EnumJson.format(CaseStatus)
-}
+enum CaseStatus:
+  case DRAFT, NEW, OPEN, SUPPRESSED, REFERRED, REJECTED, CANCELLED, SUSPENDED, COMPLETED, REVOKED, ANNULLED
+
+object CaseStatus:
+  given Format[CaseStatus] = Format(
+    Reads {
+      case JsString(s) =>
+        val statusMap = Map(
```

Comment:
Why the need for a map here? Can we not simply use "case JsString("DRAFT") => JsSuccess(CaseStatus.DRAFT)"?